homeassistant/components/knx/__init__.py

```diff
@@ -245,39 +297,44 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         schema=SERVICE_KNX_EXPOSURE_REGISTER_SCHEMA,
     )
 
-    async def reload_service_handler(service_call: ServiceCall) -> None:
-        """Remove all KNX components and load new ones from config."""
-
-        # First check for config file. If for some reason it is no longer there
-        # or knx is no longer mentioned, stop the reload.
-        config = await async_integration_yaml_config(hass, DOMAIN)
-        if not config or DOMAIN not in config:
-            return
+    return True
 
-        await asyncio.gather(
-            *(platform.async_reset() for platform in async_get_platforms(hass, DOMAIN))
-        )
-        await knx_module.xknx.stop()
 
-        await async_setup(hass, config)
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unloading the KNX platforms."""
+    knx_module: KNXModule = hass.data[DOMAIN]
+    for exposure in knx_module.exposures:
+        exposure.shutdown()
```

Comment:
Is this async friendly?

Reply:
Yes it is, it will basically just remove the registered callbacks and remove the device from the underlying library to be clean